src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -101,6 +105,7 @@ AttributeSyntax attribute
 		if( returnTypeInfo.Type.ContainingNamespace.ToString() == "System.Threading.Tasks" ) {
 			switch( returnTypeInfo.Type.MetadataName ) {
 				case "Task":
+					if( isReturnType ) { m_generatedFunctionReturnsVoid = true; }
 					return isReturnType ? SyntaxFactory.ParseTypeName( "void" ).WithTriviaFrom( typeSynt ) : typeSynt;

```

Comment:
```suggestion
					if( isReturnType ) {
						m_generatedFunctionReturnsVoid = true;
						return SyntaxFactory.ParseTypeName( "void" ).WithTriviaFrom( typeSynt )
					} else {
						return typeSynt;
					}
```

this would be simpler?